src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/workstations/Chef.java

```diff
@@ -0,0 +1,90 @@
+package com.epam.training.designpatterns.fastfoodrestaurant.workstations;
+
+import java.util.List;
+import java.util.Observable;
+import java.util.Observer;
+
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Client;
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Order;
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.ReadyMeal;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Condiment;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Food;
+import com.epam.training.designpatterns.fastfoodrestaurant.strategies.CookingStrategy;
+import com.epam.training.designpatterns.fastfoodrestaurant.strategies.DefaultCookingStrategy;
+
+public class Chef implements Observer {
+	
+	private OrderQueue orderQueue;
+	private DeliveryQueue deliveryQueue;
+	private CookingStrategy cookingStrategy = new DefaultCookingStrategy();
+	private boolean isBusy;
+	private int cookingSpeed = 1000;
+
+	public Chef(OrderQueue orderQueue, DeliveryQueue deliveryQueue) {
+		this.orderQueue = orderQueue;
+		this.deliveryQueue = deliveryQueue;
+		orderQueue.addObserver(this);
+	}
+	
+	public Chef(OrderQueue orderQueue, DeliveryQueue deliveryQueue, CookingStrategy cookingStrategy) {
+		this(orderQueue, deliveryQueue);
+		this.cookingStrategy = cookingStrategy;
+	}
+
+	@Override
+	public void update(Observable o, Object arg) {
+		if (!isBusy) {
```

Comment:
Itt átugrik rendeléseket, amikor busy, ugye? Aztán amikor megint jelez a queue, hogy jött új rendelés, akkor van esély, hogy már nem busy, és olyankor gyorsan minden rendelést kivesz a queue-ból és elkészíti az ételeket. Azon gondolkozom, lehet olyan, hogy a queue-ba kevés elem fér, egy étellel sokat foglalkozik, és a queue minden eventjét figyelmen kívül hagyja, a queue meg betelik? Olyankor a másik oldalon a pultos nem tud többet beletenni, a queue többet nem jelez, a szakács meg magától nem néz rá?

Egyelőre ne írd át, csak kérdem. :-)

Reply:
A Chef miután kapott egy notification-t a publishertől, elvileg addig veszi ki az ételeket amíg ki nem ürül a queue (while (!orderQueue.isEmpty())). Vagyis még ha sokáig is tart egy rendelés feldolgozása, és közben bekerül egy csomó új order (és ezeknek a notification-jeit már ignore-álja), akkor is felfogja dolgozni őket, mert miután végzett azzal az első hosszú rendeléssel, utána látni fogja hogy a queue nem üres, és nem lép ki a ciklusból, hanem mindig addig megy amíg ki nem üríti teljesen.